src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1118,6 +1118,15 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                 keyFileSearchPaths.Add(outputDirectory);
             }
 
+            // Public sign doesn't use the legacy search path settings
+            if (publicSign)
+            {
+                if (!string.IsNullOrWhiteSpace(keyFileSetting))
+                {
+                    keyFileSetting = ParseGenericPathToFile(keyFileSetting, diagnostics, baseDirectory);
```

Comment:
Since keyFileSearchPaths also includes outputDirectory, it probably would be appropriate to look there next.